Octokit.Tests.Integration/Helpers/OrganizationRepositoryWithTeamContext.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Octokit.Tests.Integration.Helpers
+{
+    internal class OrganizationRepositoryWithTeamContext : IDisposable
+    {
+        internal RepositoryContext RepositoryContext { get; set; }
+        internal TeamContext TeamContext { get; set; }
+
+        public void Dispose()
+        {
+            if (RepositoryContext != null)
+                RepositoryContext.Dispose();
+
+            if (TeamContext != null)
+                TeamContext.Dispose();
+        }
+    }
+
+    internal static class RepositoryProtectedBranchHelperExtensions
+    {
+        internal async static Task<RepositoryContext> CreateRepositoryWithProtectedBranch(this IGitHubClient client)
+        {
+            // Create user owned repo
+            var userRepo = new NewRepository(Helper.MakeNameWithTimestamp("protected-repo")) { AutoInit = true };
+            var contextUserRepo = await client.CreateRepositoryContext(userRepo);
+
+            // Protect master branch
+            var update = new BranchProtectionSettingsUpdate(
+                new BranchProtectionRequiredStatusChecksUpdate(true, true, new[] { "build", "test" }));
+
+            await client.Repository.Branch.UpdateBranchProtection(contextUserRepo.RepositoryOwner, contextUserRepo.RepositoryName, "master", update);
+
+            return contextUserRepo;
+        }
+
+        internal async static Task<OrganizationRepositoryWithTeamContext> CreateOrganizationRepositoryWithProtectedBranch(this IGitHubClient client)
+        {
+            // Create organization owned repo
+            var orgRepo = new NewRepository(Helper.MakeNameWithTimestamp("protected-org-repo")) { AutoInit = true };
+            var contextOrgRepo = await client.CreateRepositoryContext(Helper.Organization, orgRepo);
+
+            // Create team in org
+            var contextOrgTeam = await client.CreateTeamContext(Helper.Organization, new NewTeam(Helper.MakeNameWithTimestamp("team")));
+
+            // Grant team push access to repo
+            await client.Organization.Team.AddRepository(
+                contextOrgTeam.TeamId,
+                contextOrgRepo.RepositoryOwner,
+                contextOrgRepo.RepositoryName,
+                new RepositoryPermissionRequest(Permission.Push));
+
+            // Protect master branch
+            var orgUpdate = new BranchProtectionSettingsUpdate(
+                new BranchProtectionRequiredStatusChecksUpdate(true, true, new[] { "build", "test" }),
+                new ProtectedBranchRestrictionsUpdate(new[] { contextOrgTeam.TeamName }, new string[0]));
```

Comment:
The restrictions request object has a `string[]` for teams and a `string[]` for users so it didnt seem possible to offer a `ctor` that takes only teams and one that takes only users, since it's the same parameter signature for both.  If you have any ideas, let me know?